[v20.x backport] streams: fixes for webstreams

MattiasBuelens opened the pull request:
Manual backport of #51168 to v20.x.
Depends on #52772, so keeping this in draft until that one lands.

[This one was a bit tricky](https://github.com/nodejs/node/pull/51168#issuecomment-2088241048), since there are changes to transferables on v22.x that are not on v20.x (such as #50107) which also affect web streams. However, *I think* I made it work. 🤞

marco-ippolito commented:
can you fix the conflicts? I'll try to backport it in the next v20 release

MattiasBuelens commented:
@marco-ippolito Updated and ready for review. 😉

marco-ippolito commented:
Hi @MattiasBuelens I was backporting your commits, I encountered some issue, can you please cherry-pick the landed commits (example https://github.com/nodejs/node/pull/51168#issuecomment-1869556329) that contain metadata like PR-URL and reviewers, and please squash the fixup commits

MattiasBuelens commented:
@marco-ippolito I've squashed the commits on this PR and rebased them onto `v20.x-staging`.

The original PR didn't have the most prominent change in the first commit, so we landed it in https://github.com/nodejs/node/commit/20c63134fcd86468a940ab447bb092bdc2238389 with the unimaginative title "stream: fix code style". 😅 I changed it to "stream: fixes for webstreams" for this PR, if that's okay.

marco-ippolito commented:
> @marco-ippolito I've squashed the commits on this PR and rebased them onto `v20.x-staging`.
> 
> The original PR didn't have the most prominent change in the first commit, so we landed it in https://github.com/nodejs/node/commit/20c63134fcd86468a940ab447bb092bdc2238389 with the unimaginative title "stream: fix code style". 😅 I changed it to "stream: fixes for webstreams" for this PR, if that's okay.

You can just cherry-pick the landed commits from the original PR (at the end of the pr in the comment from the bot landed in ...) and squash only the fixup commits

mcollina reviewed (approved):
lgtm

marco-ippolito commented:
Landed in 48138afd3507
